### PR TITLE
Change read/write behavior based on friday's discussion

### PR DESCRIPTION
**Pull request by khou2020**
Read test runs first
Use read data in the write test.
Remove load data function.
Remove -R and -W. -i and -o imply write and read
Decide API to use in read test automatically.
Update README file to reflect the changes.

### CODE REVIEW

**Review by wkliao — COMMENTED** [review]
There is no need to add file extension '.nc' or '.h5' to the output file path.
This commit can be removed.

**Comment by khou2020** [comment]
> There is no need to add file extension '.nc' or '.h5' to the output file path.
> This commit can be removed.

ADIOS will always add .bp.dir. E3SM has no control with it. We add .nc and .h5 just to conform with ADIOS.

**Comment by wkliao** [comment]
But your changes are for PnetCDF and HDF5 APIs.

**Comment by wkliao** [comment]
For the input and output paths used on the command line, below are the example run commands I used.
```
# F case
mpiexec -n 16 src/e3sm_io datasets/f_case_866x72_16p.nc -k -n -o can_F_out.nc
mpiexec -n 16 src/e3sm_io datasets/f_case_866x72_16p.nc -k -n -o blob_F_out.nc -x blob -r 25
mpiexec -n 16 src/e3sm_io datasets/f_case_866x72_16p.nc -k -n -o bp_F_out -x blob -a adios -g 2

# G case
mpiexec -n 16 src/e3sm_io datasets/g_case_cmpaso_16p.nc -k -n -o can_G_out.nc
mpiexec -n 16 src/e3sm_io datasets/g_case_cmpaso_16p.nc -k -n -o blob_G_out.nc -x blob
```

The output files are
```
-rw------- 1 10166 429  16948376 Jul 11 15:20 can_F_out_h0.nc
-rw------- 1 10166 429    353840 Jul 11 15:20 can_F_out_h1.nc
-rw------- 1 10166 429   8731020 Jul 11 15:21 blob_F_out_h0.nc.0000
-rw------- 1 10166 429   8704428 Jul 11 15:21 blob_F_out_h0.nc.0001
-rw------- 1 10166 429    321136 Jul 11 15:21 blob_F_out_h1.nc.0000
-rw------- 1 10166 429    370784 Jul 11 15:21 blob_F_out_h1.nc.0001

bp_F_out_h0.bp.dir:
total 22144
-rw------- 1 10166 429 11541992 Jul 11 17:39 bp_F_out_h0.bp.0
-rw------- 1 10166 429 11126779 Jul 11 17:39 bp_F_out_h0.bp.1
-rw------- 1 10166 429     3645 Jul 11 17:39 profiling.json

bp_F_out_h1.bp.dir:
total 1244
-rw------- 1 10166 429 658529 Jul 11 17:39 bp_F_out_h1.bp.0
-rw------- 1 10166 429 609707 Jul 11 17:39 bp_F_out_h1.bp.1
-rw------- 1 10166 429   3502 Jul 11 17:39 profiling.json

-rw------- 1 10166 429 831130612 Jul 11 16:54 can_G_out.nc
-rw------- 1 10166 429 417943536 Jul 11 15:27 blob_G_out.nc.0001
-rw------- 1 10166 429 415546036 Jul 11 15:27 blob_G_out.nc.0000
```

File: README.md [hunk 1]
```diff
@@ -270,8 +287,17 @@ sharing Decomposition 4, 2 sharing Decomposition 5, and 4 sharing Decomposition
...
-       [-i path] Enable write performance evaluation and set the output file
+              If set, E3SM I/O will read the data from the input files and write it out in the write test.
+              The input files must be generated by E3SM under the same configuration (decomposition file) as the selected test case.
+              The API used in the read test is determined by the input file format.
+              The API selected (-a option) acts only as a hint to select proper VOL when reading HDF5 files.
+              When testing F case, the input files' name must be postfixed with "_h0" and "_h0" the same way as the output file.
+       [-o path] Enable write performance evaluation and set the output file
                  (folder) path
+              The file extension is determined based on the API used and should not be part of "path".
+              The F case test output 2 files. The E3SM benchmark will append "_h0" and "_h0" to them accordingly (see "Output files" section in the README file).
```
[the remark this answers is not in the document]
Comment (khou2020) [reply README.md:298]: The explanation tells the user that the benchmark will automatically add _hx. It didn't ask the user to add them.

[the remark this answers is not in the document]
Comment (wkliao) [reply README.md:298]: Move those explanation out of usage() and add them only
in README.md. You can use more space to describe options
'-i' and '-o'. Note '_h0' and '_h1' are not appended. They are
inserted before '.nc' file extension.

**Comment by khou2020** [comment]
Fixed

**Pull request merged**